### PR TITLE
Adds the M40 Adhesive grenade

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request

It is a sticky grenade.
It currently has the power of an impact grenade.

When you throw it, it sticks to things.
You cannot remove it once its stuck. It has a 5 second delay.
It wont stick to turf. walls and floors are made with no stick stuff.

## Why It's Good For The Game
This adds a new and interesting type of grenade, it is about the same power of an impact.
Each day we get closer to becoming Halo.
Nova-Epee is the one who commisioned this, they are allowed to put more reasons below.

## Changelog
:cl:
add: Adds the M40 Adhesive grenade. A sticky grenade to stick to ~~friendlies~~ xenos. Has the same power of an impact grenade and a 5 second fuse.
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
Reasons:

1. Provide marines a unique toolset. There is one tool that stick onto a xeno: using a flare and litting up xeno on fire. Interesting for sure, but niche and not a serious threat. Here is a tool that is a serious threat and the xenomorph must run away in a safe distance so they don't blow themselves up and other xenomorphs.
2. Provide xenomorphs a new way to make marines friendly fire each other. Aim mode decreases the chance of xenomorphs using marines' weapons against each other, and a funny xenomorph can bring the nade BACK to the marines. Don't be shooting these on minions that suicide charge you.
4. Halo

**Comment by Lumipharon** [comment]
The crusher charge deliberately eating sticky nades will be hilarious.

**Review by BraveMole — APPROVED** [review]
Need to clean ref on qdelete as well

**Comment by ThelastArchiver** [comment]
two important things first of all does that mean xenos get flung (in a random direction) and staggered when it explodes and second are they gonna be req grenades or free?

**Comment by Pill-U** [comment]
can xenos fling missed nades back?

**Comment by Losenis** [comment]
Finally, counterplay to GL FF. Just run up to the fucker.

**Review by Kuro020 — APPROVED** [review]

**Comment by RuneQuester** [comment]
Unavoidable grenade blasts??

**Comment by MiniMeatwad** [comment]
Make it so you can resist it off

**Comment by novaepee** [comment]
The random direction is going to get discussed with the coder.

> are they gonna be req grenades or free?

Check the code. They are free with 125 stocked.

> can xenos fling missed nades back?

If they don't stick on mobs, yes. The sticky nade doesn't stick on turf, walls, and floors, so by implication, the shrike can still play baseball with marines.

> Unavoidable grenade blasts that stagger?

You have 5 seconds to find a counterplay. If we're comparing this to impact nade, and it is important that we do, impact nade gives you no counter when you get click on. Currently, #9738 removes impact nade, and Bravemole expresses approval of sticky nade replacing impact nade.

> if it has the same power as an impact nade how is it nerfed

"the damage is delayed, so if you get crit you'll probably get crit behind allies line you can cause damage to marines you have more control over things"

"this is impact nade but xenos have more way to deal with it"

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 1]
```diff
@@ -72,6 +72,44 @@
...
+	///Current image overlay applied to stuck_to, used to remove the overlay after detonation.
+	var/image/saved_overlay
+
+/obj/item/explosive/grenade/sticky/throw_impact(atom/hit_atom, speed)
+	. = ..()
+	if(!active || stuck_to || isturf(hit_atom))
+		return
+	var/image/stuck_overlay = image(icon, hit_atom, initial(icon_state) + "_stuck")
+	stuck_overlay.pixel_x += rand(-5, 5)
+	stuck_overlay.pixel_y += rand(-7, 7)
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/marines.dm:93]: why add just set it

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 2]
```diff
@@ -72,6 +72,44 @@
...
+	hit_atom.overlays += stuck_overlay
+	forceMove(hit_atom)
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/marines.dm:95]: use the proper helpers

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by FosterGrime-Heymynameslouie** [comment]
can you theoretically stick loads of nades to a marine, charge in and blow up, essentially being a worse version of the suicide vest

**Comment by hyper2snyper** [comment]
> can you theoretically stick loads of nades to a marine, charge in and blow up, essentially being a worse version of the suicide vest

they are five seconds till detonate.
have a small amount of damage.
and can only stick when activated.
tops you get 4 stuck to you then ~2 seconds to "charge" before they blow you up and ruin your day

**Comment by TiviPlus** [comment]
Its cut not remove at least compile once smh

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 3]
```diff
@@ -72,6 +72,44 @@
...
+	hit_atom.add_overlay(stuck_overlay)
+	forceMove(hit_atom)
+	saved_overlay = stuck_overlay
+	stuck_to = hit_atom
+	RegisterSignal(stuck_to, COMSIG_PARENT_QDELETING, .proc/clean_refs)
+
+/obj/item/explosive/grenade/sticky/prime()
+	if(stuck_to)
+		stuck_to.cut_overlays(saved_overlay)
+		clean_refs()
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/marines.dm:103]:
```suggestion
		stuck_to.cut_overlay(saved_overlay)
		clean_refs()
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Pull request merged**